internal/oauth2/dummy_token.go

```diff
@@ -0,0 +1,16 @@
+package oauth2
+
+import "context"
+
+// DummyToken is a dummy implementation of the Tokener interface.
+type DummyToken struct{}
+
+// NextToken returns a static "testtoken" string.
+func (dt *DummyToken) NextToken(ctx context.Context) (string, error) {
+
+	return "testtoken", nil
+}
+
+func (dt *DummyToken) ForceRefresh(ctx context.Context) (string, error) {
+	return "", nil
+}
```

Comment:
Right, yeah, you are correct. I got lost here, because other tests in this file have very little value: they all test just the mock, they test basically no production code paths:

https://github.com/osbuild/image-builder/blob/b656bd58831787f84b0946ecd5b660d0ba93d888/internal/v1/handler_recommend_package_test.go

See `espStatusCode, body := tutils.PostResponseBody(t, apiSrv.URL, payload)`. This just test the mock defined a few lines above. If you remove `startServer` from these tests, the tests will be absolutely happy.

What shall we do about this? I suggest merging this PR "as is" (see my latest review comments), and then reverting to use `DummyToken` in tests, and also fixing the whole `handler_recommend_package_test`. (The newly added test is actually doing what's supposed to do.)

Btw, we should also add a test for the fallback behaviour to `ForceRefresh`. :)